### PR TITLE
Map layer switcher

### PR DESCRIPTION
**Pull request by geohacker**
Based on https://projects.invisionapp.com/share/CGUB4COZKTB#/screens/389450383, implements a modal to switch overalys and basemaps.

image

### CODE REVIEW

**Comment by geohacker** [comment]
This PR I think is ready:
* [x] Redesign modal
* [x] Allow switching base map
* [x] Control visibility of OSM layer

I haven't wired up the photo and traces layers yet because those aren't available on the map just yet. Will introduce that in another PR.

@batpad @sethvincent would be great if you can give this a spin!

File: app/components/BasemapModal.js [hunk 1]
```diff
@@ -24,27 +31,205 @@ const Button = styled.TouchableHighlight`
   elevation: 1;
 `
 
-export default class BasemapModal extends React.Component {
+// const CancelButton = styled.TouchableOpacity`
```
Comment (batpad) [inline app/components/BasemapModal.js]: Can we remove the commented out code before merging?

File: app/components/BasemapModal.js [hunk 2]
```diff
@@ -24,27 +31,205 @@ const Button = styled.TouchableHighlight`
...
+                  </SwitchSection>
+                  <SwitchSection>
+                    <BasemapItem onPress={() => { this.props.onChange('satellite') }}>
+                      {this.getNameStyle('satellite')}
+                    </BasemapItem>
+                  </SwitchSection>
+                </BasemapSection>
+              </View>
+            </OverlayInner>
+            {/* <CancelContainer>
```
Comment (batpad) [inline app/components/BasemapModal.js]: Commented out code.

File: app/reducers/map.js [hunk 3]
```diff
@@ -482,6 +487,14 @@ export default function (state = initialState, action) {
         baseLayer: action.baseLayer
       }
 
+    case types.TOGGLE_OVERLAY:
+      let overlays = { ...state.overlays }
+      overlays[action.layer] = !overlays[action.layer]
+      return {
+        ...state,
+        overlays: overlays
```
Comment (batpad) [inline app/reducers/map.js]: can be written slightly more concise with:

```
return {
  ...state,
  overlays
}
```

File: app/components/BasemapModal.js (hunk 1 above)
Comment (geohacker) [reply app/components/BasemapModal.js]: Ah yes this is a question I have. Do we want the close button for the modal? I want to see how this works on the device and decide if we should also add a button. If the backdrop press works we can probably ignore the cancel button.

Comment (batpad) [reply app/components/BasemapModal.js]: @geohacker clicking outside the modal to close it seemed to work well for me and, to me, seemed quite intuitive.

**Pull request merged**